### ISSUE TITLE
"block-opening-brace-newline-before" unexpectedly throws with "never-multi-line"

### ISSUE THREAD

**Issue by Nirazul** [stylelint/stylelint#1011]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

The rule throws in a multi-line block even when no newline was used.

> Which rule, if any, is this issue related to?

`block-opening-brace-newline-before`

> What CSS is needed to reproduce this issue?

e.g.

```css
a {
color: red; }
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "block-opening-brace-newline-before": "never-multi-line"
  }
}
```

> Which version of stylelint are you using?

e.g. `4.3.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

e.g. "gulp-stylelint"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
error  Unexpected whitespace before "{" of a multi-line block   block-opening-brace-newline-before
```

**Comment by davidtheclark** [comment]
This is as intended. No whitespace is allow if you use a `never` option. If you want to enforce a single space, can't you use the `space` rule?

**Comment by Nirazul** [comment]
As with the other issue report, I wasn't aware of the usage to disallow whitespace before. Sorry!

**Comment by PhilippBaschke** [comment]
I just tripped over this as well. I think the **`newline`** in `block-opening-brace-newline-before` suggest that `never-multi-line` only means that newlines are not allowed. Especially since spaces are handled by a different rule.

**Comment by davidtheclark** [comment]
I'll try to articulate the justification quick:

- If `never-*` meant "never a newline" or elsewhere meant "never a single space", corresponding exactly with what `always-*` means, then how would you outlaw tabs, or multiple spaces?
- We need a way to specify that you want *no* spaces, not just "anything but a newline".
- We think it's a very rare use case that you really want "anything but a newline" enforced. Would you really be ok with any arbitrary whitespace in that position other than a newline? 3 tabs?

**Comment by PhilippBaschke** [comment]
@davidtheclark Thank you very much for clarifying! :+1:
